modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -1245,6 +1246,52 @@ class DeConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
 
         pad.width = pad_l;
         pad.height = pad_t;
+
+        const int inpCn = inputs[0].size[1];
+        const int outCn = numOutput;
+        weightsMultipliers.assign(outCn, 1.0);
+        if (weightsMat.empty())
+        {
+            weightsMat = (blobs[0].reshape(1, inpCn)).clone();
+            biasesMat = hasBias() ? blobs[1].reshape(1, outCn) : Mat::zeros(outCn, 1, CV_32F);
+        }
+    }
+
+    virtual bool tryFuse(Ptr<Layer>& top) CV_OVERRIDE
+    {
+        Mat w, b;
+        top->getScaleShift(w, b);
+        if (!w.empty() || !b.empty())
+        {
+            fuseWeights(w, b);
+            return true;
+        }
+        return false;
+    }
```

Comment:
This method looks similar to Convolution's `tryFuse`. May I ask you remove them both and define `tryFuse` at `BaseConvolutionLayerImpl`?